src/adapters/methods/adapter_layer_base.py

```diff
@@ -103,28 +102,42 @@ def average_adapter(self, adapter_name: str, input_adapters: Dict[str, float]) -
         Returns:
             bool: True if the adapter was added, False otherwise.
         """
-        raise NotImplementedError()
+        # add new adapter
+        if self.add_adapter(adapter_name, self.layer_idx):
+            # average weights
+            avg_state_dict = {}
+            for name, weight in input_adapters.items():
+                if name in self.adapter_modules:
+                    module = self.adapter_modules[name]
+                    for k, v in module.state_dict().items():
+                        if k in avg_state_dict:
+                            avg_state_dict[k] += weight * v
+                        else:
+                            avg_state_dict[k] = weight * v
+                else:
+                    self.delete_adapter(adapter_name)  # clean up before raising error
+                    raise ValueError("Adapter {} not found.".format(name))
+            # load averaged weights
+            self.adapter_modules[adapter_name].load_state_dict(avg_state_dict)
+            return True
+
+        return False
 
-    @abstractmethod
     def delete_adapter(self, adapter_name: str):
         """Deletes an adapter module from the layer.
 
         Args:
             adapter_name (str): The name of the adapter to delete.
         """
-        raise NotImplementedError()
+        if adapter_name in self.adapter_modules:
+            del self.adapter_modules[adapter_name]
```

Comment:
Now that we have a default implementation of `delete_adapter`, `enable_adapters` and `get_adapter` you can remove the functions from the other adapter classes (LoRA, Bottleneck, ...) when they use this default implementation.
You don't have to remove the duplicate code in the classes for the `average_adapter` function as I will do this in #698

Reply:
good point, was planning to do that! now done